### PR TITLE
[DOC] Improve installation instructions

### PR DESCRIPTION
**Pull request by changhiskhan**
[SETUP] Add 'all' option to install all extras

These instructions assume we've already published to maven/pypi.
I will also need to update the dockerfile to match these instructions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,32 +69,103 @@ for example in data_loader:
...
-docker run -p 0.0.0.0:8888:8888/tcp rikai:latest jupyter lab --ip 0.0.0.0 --port 8888 --NotebookApp.quit_button=True --NotebookApp.custom_display_url=http://127.0.0.1:8888
+### <a name="Source"></a>Install from source
+
+To build from source you'll need python as well as Scala with sbt installed:
+
+```bash
+# Clone the repo
+git clone git@github.com:eto-ai/rikai rikai
+# Build the jar
+sbt publishLocal
```
Comment (eddyxu) [inline README.md:131]: i do have issue with this , to directly run with `pyspark --packages ai.eto.rikai:rikai_2.12:0.0.1` with missing netty deps. 

Lets make sure these processes getting verified automatically?

File: README.md [hunk 2]
```diff
@@ -69,32 +69,103 @@ for example in data_loader:
...
+If you want to read/write data from/to S3, you'll need to add additional options as well.
+
+Add appropriate options when creating the SparkSession:
+
+```python
+spark = (
+   SparkSession
+      .builder
+      .appName('rikai')          
+      .config('spark.jars.packages', 'ai.eto.rikai:rikai-core:0.0.1')
```
Comment (eddyxu) [inline README.md:149]: are we going to change this to `0.0.1-SNAPSHOT`?

File: README.md (hunk 1 above)
Comment (eddyxu) [reply README.md:131]:
```
java.io.FileNotFoundException: File file:/home/lei/.ivy2/jars/io.netty_netty-transport-native-epoll-4.1.46.Final.jar does not exist

	at org.apache.hadoop.fs.RawLocalFileSystem.deprecatedGetFileStatus(RawLocalFileSystem.java:611)
	at org.apache.hadoop.fs.RawLocalFileSystem.getFileLinkStatusInternal(RawLocalFileSystem.java:824)
	at org.apache.hadoop.fs.RawLocalFileSystem.getFileStatus(RawLocalFileSystem.java:601)
	at org.apache.hadoop.fs.FilterFileSystem.getFileStatus(FilterFileSystem.java:428)
	at org.apache.spark.SparkContext.addFile(SparkContext.scala:1534)
	at org.apache.spark.SparkContext.addFile(SparkContext.scala:1498)
	at org.apache.spark.SparkContext.$anonfun$new$12(SparkContext.scala:494)
	at org.apache.spark.SparkContext.$anonfun$new$12$adapted(SparkContext.scala:494)
	at scala.collection.immutable.List.foreach(List.scala:392)
```

File: README.md (hunk 2 above)
Comment (changhiskhan) [reply README.md:149]: Should just do 0.0.1 for the release?

File: README.md (hunk 1 above)
Comment (changhiskhan) [reply README.md:131]: how did you setup Spark? With pre-packaged hadoop or user provided hadoop jars?

File: README.md (hunk 2 above)
Comment (changhiskhan) [reply README.md:149]: oh you want to change this to -SNAPSHOT leading up the release? yeah that makes sense. I was planning to have a separate repo to take care of the mvn -SNAPSHOT release

**Review by eddyxu — APPROVED** [review]
LGTM

**Pull request merged**